### PR TITLE
New services for Out Of Hours Pages

### PR DESCRIPTION
**Pull request by iant**

### CODE REVIEW

**Comment by iant** [comment]
hi @ldodds this is still WIP and hasn't deviated too far from out discussion - but would be good to get an early review on this if you get a chance on Monday?  (class naming (DayTypeBreakdown & Store) and if we should re-use CombinedUsageMetric - with the addition of percent - in place of Store)

File: app/services/usage_breakdown/day_type_breakdown.rb [hunk 1]
```diff
@@ -0,0 +1,101 @@
+# frozen_string_literal: true
+
+module UsageBreakdown
+  class DayTypeBreakdown
+    attr_reader :holidays, :school_day_closed, :school_day_open, :weekends, :out_of_hours
```
Comment (iant) [inline app/services/usage_breakdown/day_type_breakdown.rb:5]: I need to also include community use here

**Comment by iant** [comment]
```
===========================================================AlertOutOfHoursElectricityUsage============================================================
{
                                                            "AlertOutOfHoursElectricityUsage:breakdown_chart" => :alert_daytype_breakdown_electricity,
                                               "AlertOutOfHoursElectricityUsage:group_by_week_day_type_chart" => :alert_group_by_week_electricity,
                                                                       "AlertOutOfHoursElectricityUsage:fuel" => "electricity",
                                                           "AlertOutOfHoursElectricityUsage:fuel_description" => "electricity",
                                                                  "AlertOutOfHoursElectricityUsage:fuel_cost" => 0.15203393721501832,
                                                          "AlertOutOfHoursElectricityUsage:fuel_cost_current" => 0.15203393721501832,
                                                             "AlertOutOfHoursElectricityUsage:total_annual_£" => 71060.41900000001,
                                                           "AlertOutOfHoursElectricityUsage:total_annual_co2" => 88492.6392,
                                                         "AlertOutOfHoursElectricityUsage:schoolday_open_kwh" => 172067.6333333333,
                                                       "AlertOutOfHoursElectricityUsage:schoolday_closed_kwh" => 181388.2666666666,
                                                               "AlertOutOfHoursElectricityUsage:holidays_kwh" => 71847.09999999999,
                                                               "AlertOutOfHoursElectricityUsage:weekends_kwh" => 42095.39999999999,
                                                              "AlertOutOfHoursElectricityUsage:community_kwh" => 0.0,
                                                           "AlertOutOfHoursElectricityUsage:total_annual_kwh" => 467398.3999999999,
                                                           "AlertOutOfHoursElectricityUsage:out_of_hours_kwh" => 295330.7666666666,
                                                     "AlertOutOfHoursElectricityUsage:schoolday_open_percent" => 0.36813911501052066,
                                                   "AlertOutOfHoursElectricityUsage:schoolday_closed_percent" => 0.3880806324254996,
                                                           "AlertOutOfHoursElectricityUsage:holidays_percent" => 0.15371704310498283,
                                                           "AlertOutOfHoursElectricityUsage:weekends_percent" => 0.09006320945899686,
                                                          "AlertOutOfHoursElectricityUsage:community_percent" => 0.0,
                                                       "AlertOutOfHoursElectricityUsage:out_of_hours_percent" => 0.6318608849894793,
                                                           "AlertOutOfHoursElectricityUsage:schoolday_open_£" => 26756.202000000005,
                                                         "AlertOutOfHoursElectricityUsage:schoolday_closed_£" => 27184.38200000001,
                                                                 "AlertOutOfHoursElectricityUsage:holidays_£" => 10813.954999999998,
                                                                 "AlertOutOfHoursElectricityUsage:weekends_£" => 6305.880000000001,
                                                                "AlertOutOfHoursElectricityUsage:community_£" => 0.0,
                                                             "AlertOutOfHoursElectricityUsage:out_of_hours_£" => 44304.217000000004,
                                                    "AlertOutOfHoursElectricityUsage:schoolday_open_£current" => 26756.202000000005,
                                                  "AlertOutOfHoursElectricityUsage:schoolday_closed_£current" => 27184.38200000001,
                                                          "AlertOutOfHoursElectricityUsage:holidays_£current" => 10813.954999999998,
                                                          "AlertOutOfHoursElectricityUsage:weekends_£current" => 6305.880000000001,
                                                         "AlertOutOfHoursElectricityUsage:community_£current" => 0.0,
                                                      "AlertOutOfHoursElectricityUsage:out_of_hours_£current" => 44304.217000000004,
                                                         "AlertOutOfHoursElectricityUsage:schoolday_open_co2" => 33246.97816666668,
                                                       "AlertOutOfHoursElectricityUsage:schoolday_closed_co2" => 35745.40483333333,
                                                               "AlertOutOfHoursElectricityUsage:holidays_co2" => 12476.783800000008,
                                                               "AlertOutOfHoursElectricityUsage:weekends_co2" => 7023.472399999997,
                                                              "AlertOutOfHoursElectricityUsage:community_co2" => 0.0,
                                                           "AlertOutOfHoursElectricityUsage:out_of_hours_co2" => 55245.66103333332,
                                              "AlertOutOfHoursElectricityUsage:good_out_of_hours_use_percent" => 0.35,
                                                            "AlertOutOfHoursElectricityUsage:average_percent" => 0.5,
                                               "AlertOutOfHoursElectricityUsage:bad_out_of_hours_use_percent" => 0.65,
                                               "AlertOutOfHoursElectricityUsage:significant_out_of_hours_use" => true,
                                            "AlertOutOfHoursElectricityUsage:percent_improvement_to_exemplar" => 0.28186088498947937,
                                                       "AlertOutOfHoursElectricityUsage:potential_saving_kwh" => 131741.32666666666,
                                                         "AlertOutOfHoursElectricityUsage:potential_saving_£" => 20029.152587063218,
                                                       "AlertOutOfHoursElectricityUsage:potential_saving_co2" => 24939.046283943266,
                                                                    "AlertOutOfHoursElectricityUsage:summary" => "£44,000pa (63% of annual cost) ",
                                      "AlertOutOfHoursElectricityUsage:tariff_has_changed_during_period_text" => "",
                                                                  "AlertOutOfHoursElectricityUsage:relevance" => :relevant,
                                                              "AlertOutOfHoursElectricityUsage:analysis_date" => #<Date: 2021-12-31 ((2459580j,0s,0n),+0s,2299161j)>,
                                                                     "AlertOutOfHoursElectricityUsage:status" => :bad,
                                                                     "AlertOutOfHoursElectricityUsage:rating" => 0.6,
                                                                       "AlertOutOfHoursElectricityUsage:term" => :longterm,
                                                               "AlertOutOfHoursElectricityUsage:bookmark_url" => "https://blog.energysparks.uk/alerts#ElectricityOutOfHours",
                                                               "AlertOutOfHoursElectricityUsage:max_asofdate" => #<Date: 2022-07-13 ((2459774j,0s,0n),+0s,2299161j)>,
                                                                     "AlertOutOfHoursElectricityUsage:pupils" => 961,
                                                                 "AlertOutOfHoursElectricityUsage:floor_area" => 5932.0,
                                                                "AlertOutOfHoursElectricityUsage:school_type" => :secondary,
                                                                "AlertOutOfHoursElectricityUsage:school_name" => "Acme Academy",
                                                     "AlertOutOfHoursElectricityUsage:school_activation_date" => nil,
                                                       "AlertOutOfHoursElectricityUsage:school_creation_date" => #<Date: 2020-10-08 ((2459131j,0s,0n),+0s,2299161j)>,
                                                                        "AlertOutOfHoursElectricityUsage:urn" => 136770,
                                                          "AlertOutOfHoursElectricityUsage:one_year_saving_£" => 20029.152587063218..20029.152587063218,
                                                      "AlertOutOfHoursElectricityUsage:one_year_saving_£_low" => 20029.152587063218,
                                                     "AlertOutOfHoursElectricityUsage:one_year_saving_£_high" => 20029.152587063218,
                                                        "AlertOutOfHoursElectricityUsage:one_year_saving_co2" => 24939.046283943266,
                                                        "AlertOutOfHoursElectricityUsage:ten_year_saving_co2" => 249390.46283943264,
                                                  "AlertOutOfHoursElectricityUsage:average_one_year_saving_£" => 20029.152587063218,
                                                  "AlertOutOfHoursElectricityUsage:average_ten_year_saving_£" => 200291.52587063218,
                                                          "AlertOutOfHoursElectricityUsage:ten_year_saving_£" => 200291.52587063218..200291.52587063218,
                                                      "AlertOutOfHoursElectricityUsage:ten_year_saving_£_low" => 200291.52587063218,
                                                     "AlertOutOfHoursElectricityUsage:ten_year_saving_£_high" => 200291.52587063218,
                                                              "AlertOutOfHoursElectricityUsage:payback_years" => nil,
                                                      "AlertOutOfHoursElectricityUsage:average_payback_years" => 0.0,
                                                               "AlertOutOfHoursElectricityUsage:capital_cost" => nil,
                                                       "AlertOutOfHoursElectricityUsage:average_capital_cost" => 0.0,
                                                                  "AlertOutOfHoursElectricityUsage:timescale" => "last year",
                                                     "AlertOutOfHoursElectricityUsage:time_of_year_relevance" => 5.0,
                                "AlertOutOfHoursElectricityUsage:AlertOutOfHoursElectricityUsage:Holiday:kWh" => 71847.09999999999,
                                  "AlertOutOfHoursElectricityUsage:AlertOutOfHoursElectricityUsage:Holiday:£" => 10813.954999999998,
                             "AlertOutOfHoursElectricityUsage:AlertOutOfHoursElectricityUsage:Holiday:CO2 kg" => 12476.783800000008,
                            "AlertOutOfHoursElectricityUsage:AlertOutOfHoursElectricityUsage:Holiday:Percent" => 0.15371704310498283,
                                "AlertOutOfHoursElectricityUsage:AlertOutOfHoursElectricityUsage:Weekend:kWh" => 42095.39999999999,
                                  "AlertOutOfHoursElectricityUsage:AlertOutOfHoursElectricityUsage:Weekend:£" => 6305.880000000001,
                             "AlertOutOfHoursElectricityUsage:AlertOutOfHoursElectricityUsage:Weekend:CO2 kg" => 7023.472399999997,
                            "AlertOutOfHoursElectricityUsage:AlertOutOfHoursElectricityUsage:Weekend:Percent" => 0.09006320945899686,
                        "AlertOutOfHoursElectricityUsage:AlertOutOfHoursElectricityUsage:School Day Open:kWh" => 172067.6333333333,
                          "AlertOutOfHoursElectricityUsage:AlertOutOfHoursElectricityUsage:School Day Open:£" => 26756.202000000005,
                     "AlertOutOfHoursElectricityUsage:AlertOutOfHoursElectricityUsage:School Day Open:CO2 kg" => 33246.97816666668,
                    "AlertOutOfHoursElectricityUsage:AlertOutOfHoursElectricityUsage:School Day Open:Percent" => 0.36813911501052066,
                      "AlertOutOfHoursElectricityUsage:AlertOutOfHoursElectricityUsage:School Day Closed:kWh" => 181388.2666666666,
                        "AlertOutOfHoursElectricityUsage:AlertOutOfHoursElectricityUsage:School Day Closed:£" => 27184.38200000001,
                   "AlertOutOfHoursElectricityUsage:AlertOutOfHoursElectricityUsage:School Day Closed:CO2 kg" => 35745.40483333333,
                  "AlertOutOfHoursElectricityUsage:AlertOutOfHoursElectricityUsage:School Day Closed:Percent" => 0.3880806324254996,
              "AlertOutOfHoursElectricityUsage:AlertOutOfHoursElectricityUsage:Holiday:£ (at current tariff)" => 10813.954999999998,
              "AlertOutOfHoursElectricityUsage:AlertOutOfHoursElectricityUsage:Weekend:£ (at current tariff)" => 6305.880000000001,
      "AlertOutOfHoursElectricityUsage:AlertOutOfHoursElectricityUsage:School Day Open:£ (at current tariff)" => 26756.202000000005,
    "AlertOutOfHoursElectricityUsage:AlertOutOfHoursElectricityUsage:School Day Closed:£ (at current tariff)" => 27184.38200000001
```

File: app/services/usage_breakdown/store.rb [hunk 2]
```diff
@@ -0,0 +1,13 @@
+# frozen_string_literal: true
+
+module UsageBreakdown
+  class Store
```
Comment (iant) [inline app/services/usage_breakdown/store.rb:4]: use CombinedUsageMetric here and add a percentage accessor 
`  def initialize(kwh:, £:, co2:, percent: nil)`

**Review by ldodds — APPROVED** [review]

**Pull request merged**